extensions/nextdns/src/components/add-domain.tsx

```diff
@@ -0,0 +1,63 @@
+import { Action, ActionPanel, Form, Icon, showToast, Toast, useNavigation } from "@raycast/api";
+import { useForm } from "@raycast/utils";
+import isValidDomain from "is-valid-domain";
+import { DomainSubmitValues, Mutate } from "../types";
+import { addDomain } from "../libs/api";
+
+export default function AddDomain(props: { type: string; mutate?: Mutate | undefined }) {
+  const { pop } = useNavigation();
+  const { type, mutate } = props;
+  const description =
+    type === "allow"
+      ? "Allowing a domain will automatically allow all its subdomains. Allowing takes precedence over everything else, including security features."
+      : "Denying a domain will automatically deny all its subdomains.";
+  const { itemProps, handleSubmit } = useForm<DomainSubmitValues>({
+    async onSubmit(values) {
+      if (mutate) {
+        const toast = await showToast({ style: Toast.Style.Animated, title: "Adding Domain", message: values.domain });
+        try {
+          await mutate(addDomain({ domain: values.domain, type }), {
+            optimisticUpdate(data) {
+              const { result, profileName } = data;
+
+              if (result.some((item) => item.id === values.domain)) {
+                toast.style = Toast.Style.Failure;
+                toast.title = "Domain already added";
+              } else {
+                result.unshift({ id: values.domain, type, active: true });
+              }
+
+              return { result, profileName };
+            },
+          });
+          toast.style = Toast.Style.Success;
+          toast.title = "Added Domain";
+          toast.message = values.domain;
+          pop();
```

Comment:
(non-blocking) Not sure if we should use success toast in `.then()` clause of mutate Promise and pop in `.finally()` clause. This also works but I am thinking about avoiding any race-conditions. Same goes for other mutates. Something like this: 

https://github.com/raycast/extensions/blob/b8778e0dd272b7dba9767de6d6f0037e3577a36e/extensions/linear/src/components/docs/DocumentActions.tsx#L151-L156

Reply:
This should be fine, leaving for now 🙂

Reply:
Nvm this what you have right now works fine. You can resolve this.